src/Microsoft.ML.Data/Transforms/ConcatTransform.cs

```diff
@@ -527,6 +527,19 @@ private static VersionInfo GetVersionInfo()
 
         public override ISchema Schema => _bindings;
 
+        public ConcatTransform(IHostEnvironment env, IDataView input, string outputColumn, params string[] inputColumns)
+            : base(env, RegistrationName, input)
+        {
+            var cols = new Column[1];
+            cols[0] = new Column()
+            {
+                Name = outputColumn,
+                Source = inputColumns
+            };
+
+            _bindings = new Bindings(cols, null, Source.Schema);
```

Comment:
>_bindings = new Bindings(cols, null, Source.Schema); [](start = 12, length = 52)

I'd prefer that you construct an arguments object, then pass it via `: this(env, args, input)`. The reason is, you have missed some validation that happens -- e.g., you have missed the checks that currently appear in the alternate constructor on lines 551-552... I do not suggest duplicating them, but calling them.

Since you cannot call a constructor beyond that call AFAIK, this implies that you will probably need some sort of `private static Arguments` utility method to create that `Arguments` object, so you can pass it in. Or, perhaps even better, add a convenience constructor on the `Arguments` object itself, to hold this logic.

Generally I feel like if the constructors here do *anything at all*, the scheme is probably not well designed, since that means that there is probably some code and logic duplication going on. (E.g., what we see between lines 540 and 554.)